_posts/2022-05-19-theory-of-mind.md

```diff
@@ -0,0 +1,144 @@
+
+# Theory of Mind
+
+Whenever we converse, we take into account what we expect the other person to understand through our words as well as their possible responses. The ability to conceive such "theories" of other participants' mental states is termed as the Theory of Mind (ToM).
+
+Having ToM requires the agent to acknowledge the fact that others (including the agent itself) can _believe_ in things which are not true. These beliefs are called _false beliefs_. An agent possessing ToM can identify their own as well as other's false beliefs and take actions to confirm and hence correct these false-beliefs.
+
+# What makes a conversation _human-like_?
+
+Proposition: 
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _A dialogue is human-like if both agents participating have some degree of Theory of Mind._
+
+Theory of Mind is not limited to the content of the speech (such as the words spoken), but also addresses the mannerism of speech (prosody), facial and other non-verbal cues etc. It is easy to see that if any one of the agents lack ToM or have a poor ability, the conversation becomes uncomfortable and frustrating.
+
+However, Theory of Mind is an acquired skill, expertise of humans on ToM matures over the lifespan [2], in-addition to depending on the amount of socialization the person part-takes in. This makes quantifying the degree of expertise over ToM difficult, hence quantifying the degree of _human-likeness_ is also difficult, in-addition to being be subjective. 
+
+## Testing the Presence of Theory of Mind
+
+Testing whether or not an agent is capable of modeling mental states of others is important for many reasons, one such application is diagonosing mental disorders. Such tests are called false-belief tasks. These tests check whether the agent can model other's false-beliefs and/or confirm and correct its own false-beliefs. We will discuss two popular false-belief tasks: "Sally-Anne" and "Smarties" tasks .
+
+### Sally-Anne Task
+![Illustration of the "Sally-Anne" Test](https://upload.wikimedia.org/wikipedia/en/a/ac/Sally-Anne_test.jpg)
+
+The participating agent is told the following scenario:
+1. Sally and Anne are inside a room. 
+2. Sally has a basket with one marble inside it.
+3. Anne has an empty box with her.
+4. Sally leaves the room without her basket.
+5. Anne takes the marble out of Sally's basket and puts it in her own box.
+6. Sally comes back inside the room.
+
+Now, the participant is asked, "Where will Sally look for her marble?". If the participant replies with "the basket", this means that the participant is able to model the mental state of a fictional character Sally, and that she doesn't know that Anne took her marble. Children below the age of 3-4 answer with "the box", however, older children answer with "the basket". Some children with mental disabilities such as Down syndrome and Autism are unable to pass this test.
```

Comment:
pass/fail is somewhat unclear here. The first time I read this para, I thought both answers could be correct is what we are going for.

Reply:
Sally was outside the room when Anne swapped the place of the marble. According to Sally, the marble should be in her basket. However, the child (the participant) knows that the marble has been swapped. Agents who don't have a theory of mind _cannot_ model the right state of mind for Sally, instead, they say Sally will look for her marble in Anne's box first.

Does this make sense now?